src/main/java/com/bootcamp/demo/restapi/exceptionhandler/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,115 @@
+    }
+
+    @ExceptionHandler(HttpRequestMethodNotSupportedException.class)
+    private ResponseEntity<ApiError> handleHttpRequestMethodNotSupported(HttpRequestMethodNotSupportedException ex) {
+        String errorMessage = ex.getMethod() + " method not supported";
+        HttpStatus status = HttpStatus.BAD_REQUEST;
+        return handleExceptionInternally(new ApiError(errorMessage, status.value()), new HttpHeaders(), status);
+    }
+
+    @ExceptionHandler(HttpMessageNotReadableException.class)
+    private ResponseEntity<ApiError> handleHttpMessageNotReadableException() {
+        String errorMessage = "Json couldn't been parsed.";
+        HttpStatus status = HttpStatus.BAD_REQUEST;
+        return handleExceptionInternally(new ApiError(errorMessage, status.value()), new HttpHeaders(), status);
+    }
+
+    @ExceptionHandler(TypeMismatchException.class)
+    private ResponseEntity<ApiError> handleTypeMismatchException(TypeMismatchException ex) {
+        String errorMessage = ex.getValue() + " has invalid type.";
+        HttpStatus status = HttpStatus.BAD_REQUEST;
+        return handleExceptionInternally(new ApiError(errorMessage, status.value()), new HttpHeaders(), status);
+    }
+
+    @ExceptionHandler(MethodArgumentTypeMismatchException.class)
+    private ResponseEntity<ApiError> handleMethodArgumentTypeMismatchException(MethodArgumentTypeMismatchException ex) {
+        String errorMessage = ex.getValue() + " has invalid type.";
+        HttpStatus status = HttpStatus.BAD_REQUEST;
+        return handleExceptionInternally(new ApiError(errorMessage, status.value()), new HttpHeaders(), status);
+    }
+
+    @ExceptionHandler(Exception.class)
+    private ResponseEntity<ApiError> handleException(Exception ex) {
+        // Converting the stacktrace to string and writing to the log file
+        LOGGER.error(ExceptionUtils.getStackTrace(ex));
+        String errorMessage = "Oops. Something went wrong.";
+        HttpStatus status = HttpStatus.INTERNAL_SERVER_ERROR;
+        return handleExceptionInternally(new ApiError(errorMessage, status.value()), new HttpHeaders(), status);
+    }
+
+    /*
+        Custom exception handlers
+    */
+
+    @ExceptionHandler(ItemNotFoundException.class)
+    private ResponseEntity<ApiError> handleItemNotFoundException(ItemNotFoundException ex) {
+        String errorMessage = ex.getMessage().equals("") ? "Failed to find item" : ex.getMessage();
+        HttpStatus status = HttpStatus.NOT_FOUND;
+        return handleExceptionInternally(new ApiError(errorMessage, status.value()), new HttpHeaders(), status);
+    }
+
+    @ExceptionHandler(ServiceException.class)
+    private ResponseEntity<ApiError> handleServiceException(ServiceException ex) {
+        String errorMessage = "Oops. Something went wrong";
+        HttpStatus status = HttpStatus.INTERNAL_SERVER_ERROR;
+        return handleExceptionInternally(new ApiError(errorMessage, status.value()), new HttpHeaders(), status);
```

Comment:
Since you always send  new HttpHeaders() you could use
`public ResponseEntity(@Nullable T body, HttpStatus status)` constructor.